doc/how_to/process_by_channel_group.rst

```diff
@@ -0,0 +1,216 @@
+Processing a Recording by Channel Group
+=======================================
+
+In this tutorial, we will walk through how to preprocess and sort a recording
+separately for *channel groups*. A channel group is a subset of channels grouped by some
+feature - for example a multi-shank Neuropixels recording in which the channels
+are grouped by shank.
+
+**Why preprocess by channel group?**
+
+Certain preprocessing steps depend on the spatial arrangement of the channels.
+For example, common average referencing (CAR) averages over channels (separately for each time point)
+and subtracts the average. In such a scenario it may make sense to group channels so that
+this averaging is performed only over spatially close channel groups.
+
+**Why sort by channel group?**
+
+When sorting, we may want to completely separately channel groups so we can
+consider their signals in isolation. If recording from a long
+silicon probe, we might want to sort different brain areas separately,
+for example using a different sorter for the hippocampus, the thalamus, or the cerebellum.
+
+
+Splitting a Recording by Channel Group
+--------------------------------------
+
+In this example, we create a 384-channel recording with 4 shanks. However this could
+be any recording in which the channel are grouped in some way, for example
+a multi-tetrode recording with channel groups representing the channels on each individual tetrodes.
+
+First, let's import the parts of SpikeInterface we need into Python, and generate our toy recording:
+
+.. code-block:: python
+
+    import spikeinterface.extractors as se
+    import spikeinterface.preprocessing as spre
+    from spikeinterface import aggregate_channels
+    from probeinterface import generate_tetrode, ProbeGroup
+    import numpy as np
+
+    # Create a toy 384 channel recording with 4 shanks (each shank contain 96 channels)
+    recording, _ = se.toy_example(duration=[1.00], num_segments=1, num_channels=384)
```

Comment:
Open question:
When do we use `toy_example` and we do we use `generate_recording` or `generate_ground_truth_recording`.  I personally would prefer to only use the latter but maybe it does not matter.

Reply:
I am not sure on this and will defer to @alejoe91 @samuelgarcia

Reply:
I think that for docs `toy_example` is ok! It's actually easier to intepret